### PR TITLE
feat(compact): optimize compact for data load.

### PR DESCRIPTION
**Pull request by youngsofun**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

an improved and simpler version of https://github.com/datafuselabs/databend/issues/8311

simply resize to 1 and resize back to max_threads is enough: new BlockCompactorNoSplit make sure big block pass through it to downstream(Sink) fast.

The premise is that  DeserializerProcessor in multi threads already try to accumulate big Block, before output to the compactor.

other optimize:
-   call block.memory_size() only once.
-   avoid call concat_blocks for each new block

refactor:

- get real BlockCompactThresholds  from dest table

Closes https://github.com/datafuselabs/databend/issues/8311

### CODE REVIEW

**Comment by youngsofun** [comment]
this also makes it easier if we want to copy to commit by file later.
because we can maintain blocks for all reading files in one map.

**Comment by BohuTANG** [comment]
Some issues with this PR need to be addressed:
```
Following failed statements:
---------------------------------------------
Runner: mysql
ErrorType: statement ok execute with exception
Message: Failed to execute. Collected info: 1105 (HY000): Code: 1068, displayText = Cannot join handle from context's runtime, cause: task 14675 panicked.

Parsed Statement
    at_line: 22,
    s_type: Statement: ok, type: None,
    suite_name: base/03_dml/03_0028_copy_into_stage,
    text:
        copy into test_table from @test;
    results: [],
    runs_on: {'mysql', 'clickhouse', 'http'},
---------------------------------------------
Runner: mysql
ErrorType: statement ok execute with exception
Message: Failed to execute. Collected info: 1105 (HY000): Code: 1068, displayText = Cannot join handle from context's runtime, cause: task 15186 panicked.
```

https://github.com/datafuselabs/databend/actions/runs/3398563740/jobs/5651782219#step:4:2092

File: src/query/catalog/src/table.rs [hunk 1]
```diff
@@ -162,6 +163,7 @@ pub trait Table: Sync + Send {
         ctx: Arc<dyn TableContext>,
         pipeline: &mut Pipeline,
         need_output: bool,
+        _is_ingest: bool,
```
Comment (BohuTANG) [inline src/query/catalog/src/table.rs:166]: Prefer to add an argument instead the `is_ingest`:
```
pub enum AppendMode {
    // From INSERT operation
    Normal,
    // From COPY operation
    Copy,
}
```

Comment (youngsofun) [reply src/query/catalog/src/table.rs]: done

**Comment by youngsofun** [comment]
the effect is demonstrated in the test. https://github.com/datafuselabs/databend/pull/8644/commits/e0dee7f3113230144c3036f1fdd415330caea861

**Comment by BohuTANG** [comment]
From my `COPY`  test, this PR works well 👍 :
image

**Review by BohuTANG — APPROVED** [review]

**Pull request merged**